src/transformers/modeling_utils.py

```diff
@@ -3230,32 +2983,16 @@ def from_pretrained(
                 )
             else:
                 max_memory = get_max_memory(max_memory)
-            if getattr(model, "quantization_method", None) == QuantizationMethod.BITS_AND_BYTES:
-                # need more space for buffers that are created during quantization
-                max_memory = {key: val * 0.90 for key, val in max_memory.items()}
+            if quantizer is not None:
+                max_memory = quantizer.adjust_max_memory(max_memory)
             device_map_kwargs["max_memory"] = max_memory
 
             # Make sure tied weights are tied before creating the device map.
             model.tie_weights()
             device_map = infer_auto_device_map(model, dtype=target_dtype, **device_map_kwargs)
 
-            if load_in_8bit or load_in_4bit:
-                # The LM head / tied weights or any last module can stay on disk / CPU
-                device_map_without_lm_head = {
-                    key: device_map[key] for key in device_map.keys() if key not in modules_to_not_convert
-                }
-                if "cpu" in device_map_without_lm_head.values() or "disk" in device_map_without_lm_head.values():
-                    raise ValueError(
-                        """
-                        Some modules are dispatched on the CPU or the disk. Make sure you have enough GPU RAM to fit
-                        the quantized model. If you want to dispatch the model on the CPU or the disk while keeping
-                        these modules in 32-bit, you need to set `load_in_8bit_fp32_cpu_offload=True` and pass a custom
-                        `device_map` to `from_pretrained`. Check
-                        https://huggingface.co/docs/transformers/main/en/main_classes/quantization#offload-between-cpu-and-gpu
-                        for more details.
-                        """
-                    )
-                del device_map_without_lm_head
+            if quantizer is not None:
+                quantizer.validate_device_map
```

Comment:
Then this should be : 

```suggestion
                quantizer.validate_device_map(device_map)
```

No?